strategy-offline-fallback/controlled.js

```diff
@@ -0,0 +1,18 @@
+// To be able to provide fallbacks since the beginning we can only load the
+// image once we know the SW is ready. In addition, the SW should start
+// intercepting ASAP.
+if (navigator.serviceWorker.controller) {
```

Comment:
why are you checking for the `controller` property? how does it work? this is a bit confusing